### PR TITLE
base1: fix some test-http asserts

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
Using `assert.async()` is a bit dangerous with QUnit: you need to make
sure the `done()` method always gets called.

We have a few cases in our tests that fail when run against the Python
bridge, but don't call `done()`.  Clean those up.

In general, converting the tests to async/await syntax is easier than
trying to sort out the state machines involved here.

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
This is the set of fixes required to get "http" out of the "hanging" column into the "failing" column on the python branch.

File: pkg/base1/test-http.js [hunk 1]
```diff
@@ -284,102 +284,47 @@ QUnit.test("http promise recursive", function (assert) {
...
-                    ex.problem = "not-found";
-                assert.strictEqual(ex.problem, "not-found", "can't resolve is not found");
-            })
-            .always(function() {
-                assert.equal(this.state(), "rejected", "should fail");
-                done();
-            });
+    assert.rejects(cockpit.http({ port: 8080, address: "the-other-host.example.com" }).get("/"),
+                   /* Unfortunately we can see either of these errors when running unit tests */
+                   ex => { return ex.problem === "timeout" || ex.problem === "not-found" });
```
Comment (martinpitt) [inline pkg/base1/test-http.js:327]: Style guide: Drop useless `{ return .. }` wrappers for arrows, to stay functional. I.e. `ex => ex.problem === "timeout" || ex.problem === "not-found"`.

Not a reason to block this, though.

**Review by martinpitt — APPROVED** [review]
Thanks, this looks really nice!

Also, for future PRs: no-test please.

File: pkg/base1/test-http.js (hunk 1 above)
Comment (allisonkarlitskaya) [reply pkg/base1/test-http.js:327]: Oof.  I tried to translate it more literally first, which is why I did that.  Oh well.

**Pull request merged**